.github/workflows/changed_solutions.yml

```diff
@@ -0,0 +1,61 @@
+name: Changed Solutions
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.actor, '[bot]') != true }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "npm"
+      - run: npm ci
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v22
+      - name: Run solutions in changed files directory
+        run: |
+          root_dir=$(pwd)
+          failed_tests=()
+
+          function run_tests {
+            echo -e "\nRunning test for $1"
+            cd $1
+            npm run test:solutions || failed_tests+=("$1")
+            cd $root_dir
+            return 0
+          }
+
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            dir=$(dirname $file)
+            if [ -e $dir/package.json ]; then
+              run_tests $dir
+            elif [ -e $dir/solution.ts ]; then
+              run_tests $dir
+            elif [ -e $dir/solution.js ]; then
```

Comment:
The two `elif`s can be joined into the first `if`: https://unix.stackexchange.com/questions/47584/in-a-bash-script-using-the-conditional-or-in-an-if-statement. Let's do that to reduce the amount of duplicated/extra code.

Reply:
wowwwwwwww 🥺, I did not know this was possible.. It looks so dope !!!

Reply:
Thanks for elightening me on this @JoshuaKGoldberg ✨